docs/api_reference/src/quantities.tex

```diff
@@ -5416,119 +5651,1485 @@
+namespace mp_units {
+template<@\libconcept{Quantity}@ Q>
+  requires std::@\stdconcept{floating_point}@<typename Q::rep>
+struct exponential_distribution                                                         // hosted
+    : public std::exponential_distribution<typename Q::rep> {
+  using rep = Q::rep;
+  using base = std::exponential_distribution<rep>;
+
+  exponential_distribution();
+  explicit exponential_distribution(const rep& lambda);
+
+  template<typename Generator>
+  Q operator()(Generator& g);
+
+  Q min() const;
+  Q max() const;
+};
+}
+\end{codeblock}
+
+\rSec4[qty.rand.pois.gamma]{Class template \tcode{gamma_distribution}}
+
+\indexlibraryglobal{gamma_distribution}
+\begin{codeblock}
+namespace mp_units {
+template<@\libconcept{Quantity}@ Q>
+  requires std::@\stdconcept{floating_point}@<typename Q::rep>
+struct gamma_distribution                                                               // hosted
+    : public std::gamma_distribution<typename Q::rep> {
+  using rep = Q::rep;
+  using base = std::gamma_distribution<rep>;
+
+  gamma_distribution();
+  gamma_distribution(const rep& alpha, const rep& beta);
+
+  template<typename Generator>
+  Q operator()(Generator& g);
+
+  Q min() const;
+  Q max() const;
+};
+}
+\end{codeblock}
+
+\rSec4[qty.rand.pois.weibull]{Class template \tcode{weibull_distribution}}
+
+\indexlibraryglobal{weibull_distribution}
+\begin{codeblock}
+namespace mp_units {
+template<@\libconcept{Quantity}@ Q>
+  requires std::@\stdconcept{floating_point}@<typename Q::rep>
+struct weibull_distribution                                                             // hosted
+    : public std::weibull_distribution<typename Q::rep> {
+  using rep = Q::rep;
+  using base = std::weibull_distribution<rep>;
+
+  weibull_distribution();
+  weibull_distribution(const rep& a, const rep& b);
+
+  template<typename Generator>
+  Q operator()(Generator& g);
+
+  Q min() const;
+  Q max() const;
+};
+}
+\end{codeblock}
+
+\rSec4[qty.rand.pois.extreme]{Class template \tcode{extreme_value_distribution}}
+
+\indexlibraryglobal{extreme_value_distribution}
+\begin{codeblock}
+namespace mp_units {
+template<@\libconcept{Quantity}@ Q>
+  requires std::@\stdconcept{floating_point}@<typename Q::rep>
+struct extreme_value_distribution                                                       // hosted
+    : public std::extreme_value_distribution<typename Q::rep> {
+  using rep = Q::rep;
+  using base = std::extreme_value_distribution<rep>;
+
+  extreme_value_distribution();
+  extreme_value_distribution(const Q& a, const rep& b);
+
+  template<typename Generator>
+  Q operator()(Generator& g);
+
+  Q a() const;
+
+  Q min() const;
+  Q max() const;
+};
+}
+\end{codeblock}
+
+\indexlibrarymemberx{extreme_value_distribution}{operator()}
+\begin{itemdecl}
+template<typename Generator>
+Q operator()(Generator& g);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to:
+\tcode{return Q(base::operator()(g));}
+\end{itemdescr}
```

Comment:
Can I get rid of these optimizations?
The requirements should be equivalent.
However, I don't see that `Q::unit` is required to be `mp_units::one`.

Reply:
One of our contributors provided this code. I did a short review but never had time to analyze all the generators properly. If you see any issues, please do not hesitate to fix them or create a new Issue with the list of things to do.